tests/test_cases/test_array_simple/test_array_simple.py

```diff
@@ -0,0 +1,197 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Test getting and setting values of arrays"""
+import contextlib
+import logging
+
+import cocotb
+from cocotb.binary import BinaryValue
+from cocotb.clock import Clock
+from cocotb.result import TestFailure
+from cocotb.triggers import Timer
+
+tlog = logging.getLogger("cocotb.test")
+
+def _check_value(tlog, hdl, expected):
+    if hdl.value != expected:
+        raise TestFailure("{2!r}: Expected >{0}< but got >{1}<".format(expected, hdl.value, hdl))
```

Comment:
Nit: The 2-0-1 order is unnecessary here, or is there a deeper meaning that I'm missing?
This is also in ``test_array.py``.

Reply:
I just copied the from the other `_check_*` functions for consistency.